src/components/sockend.js

```diff
@@ -84,31 +84,45 @@ module.exports = class Sockend extends Configurable(Component) {
             publisherNamespaces['/' + normalizedNamespace] = true;
             obj.namespace = namespace;
 
+            const broadcasts = new Set(obj.advertisement.broadcasts);
+
             const subscriber = new Subscriber({
                 name: 'sockendSub',
                 namespace: namespace,
                 key: originalKey,
-                subscribesTo: obj.advertisement.broadcasts,
+                subscribesTo: ['*'],
             }, discoveryOptions);
 
-            subscriber.onMonitorAdded = () => { };
+            subscriber.onMonitorAdded = () => {
+            };
 
             obj.subscriber = subscriber;
 
-            subscriber.on('**', function(data) {
+            subscriber.on('**', function (data) {
                 if (this.event == 'cote:added' || this.event == 'cote:removed') return;
 
                 let topic = this.event.split('::');
                 let namespace = '';
 
+                let room;
                 if (topic.length > 1) {
-                    namespace += '/' + topic[0];
+                    if (topic[0].startsWith('#')) {
+                        room = topic[0].replace('#', '');
+                    } else {
+                        namespace += '/' + topic[0];
+                    }
```

Comment:
This is where the core logic should be, and therefore, change. It should support `(namespace::)(#room::)(category::)topic` and the ones in parentheses are optional. I have a rudimentary solution like the following, but if you can come up with a better version, you are welcome:

```
let topic = this.event.split('::');
let namespace = '';
let room;

if (topic.length > 1) {
    if (!topic[0].startsWith('#')) {
        namespace += '/' + topic[0];
        topic = topic.slice(1);
    }

    if (topic[0].startsWith('#')) {
        room = topic[0].slice(1);
        topic = topic.slice(1);
    }
}

topic = topic.join('');
```